vendors/halo2/README.md

```diff
@@ -0,0 +1,138 @@
+# Halo2
+
+## How to migrate
+
+Tachyon can replace the existing Halo2 implementations to generate proofs and currently supports *BN256 (or Bn254)* curve along with *GWC* and *SHPlonk* provers.
+
+### Requirements
+
+1. Replace `halo2` library in your program with [tachyon-halo2](https://github.com/kroma-network/halo2).
+
+   ``` diff
+   # In your Cargo.toml
+   - halo2_proofs = { git = "https://github.com/zcash/halo2.git" }
+   + halo2_proofs = { git = "https://github.com/kroma-network/halo2.git", branch = "halo2-with-tachyon" }
+   ```
+
+2. Install the tachyon debian package, See [Debian packaging](../../docs/how_to_use/how_to_build.md#debian-packaging).
+
+### Migration Steps
```

Comment:
```suggestion
### Steps to make proof generation Tachyon-compatible
```
or a similar feeling of we're trying to edit proof generation steps to work for tachyon